File: csharp/P0001_TwoSum/Solutions/BruteForce.cs
// Time complexity: O(n^2)
// Space complexity: O(n)

namespace P0001_TwoSum.Solutions.BruteForce;

public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        // Iterate to the penultimate index. The last index will be j
        for (int i = 0; i < nums.Length - 1; i++)
        {
            int complement = target - nums[i];

            for (int j = i + 1; j < nums.Length; j++)
            {
                if (nums[j] == complement)
                    return [i, j];
            }
        }

        // This should be unreachable
        return [-1];
    }
}

File: rust/p0217_contains_duplicate/src/brute_force.rs
pub struct Solution;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        for i in 0..nums.len() {
            for j in i + 1..nums.len() {
                if nums[i] == nums[j] {
                    return true;
                }
            }
        }
        false
    }
}
